vllm/v1/worker/tpu_model_runner.py

```diff
@@ -746,13 +768,41 @@ def capture_model(self) -> None:
         start = time.perf_counter()
         num_tokens = 16
         while True:
-            self._dummy_run(self.kv_caches, num_tokens)
             logger.info("  -- num_tokens: %d", num_tokens)
+            self._dummy_run(self.kv_caches, num_tokens)
             xm.mark_step()
-            xm.wait_device_ops()
-            if num_tokens >= self.max_num_tokens:
+            if num_tokens >= self.scheduler_config.max_num_batched_tokens:
```

Comment:
Actually I think we should use `self.max_num_tokens` still here since I made this be the padded number of tokens, in case the user put in an odd number or something weird